### ISSUE TITLE
Error Connect Query in Server TS

### ISSUE THREAD

**Issue by Delandilucas** [TheHolyWaffle/TeamSpeak-3-Java-API#113]
Hello, little time ago I had made a topic with the same error, that tried to help me and closed the topic, but has not yet been resolved. I forgot to put the code, but it is with the door too, so my question!

Error:

[2016-07-15 17:26:31.108] > whoami
[2016-07-15 17:26:31.523] [whoami] < virtualserver_status=unknown virtualserver_unique_identifier virtualserver_port=0 virtualserver_id=0 client_id=0 client_channel_id=0 client_nickname client_database_id=1 client_login_name=serveradmin client_unique_identifier=serveradmin client_origin_server_id=0
[2016-07-15 17:26:31.763] [whoami] < error id=0 msg=ok

Code:

                final TS3Config config = new TS3Config();
		config.setHost("ts.wgbrasil.tk");
		config.setDebugLevel(Level.ALL);
		
		final TS3Query query = new TS3Query(config);
		query.connect();
		
		final TS3Api api = new TS3Api(query);
		api.login("serveradmin", "minhasenha");
		api.selectVirtualServerByPort(9987);
		api.setNickname("MushTest");
		api.sendChannelMessage("MushTest está online!!");

Help-me

**Comment by Firedroide** [comment]
You could have just re-opened your last issue (#112) or commented that that answer didn't actually address your problem - but whatever, this works, too :stuck_out_tongue: 

On your last issue, the log included this strange line:

```
[2016-07-14 19:30:56.327] [login] < error id=0 msg=ok
[2016-07-14 19:30:56.651] > use                                                            <---
[2016-07-14 19:30:57.067] [use] < error id=0 msg=ok
[2016-07-14 19:30:57.352] > clientupdate client_nickname=MushTest
```

Both me and @mohe2015 concluded that you weren't using `selectVirtualServerByPort` or `selectVirtualServerById` correctly.
`api.selectVirtualServerById(-1)` would produce this output, for example.

---

Now your log shows that when you run `whoAmI()` you're still not on a virtual server, despite the code showing a correct invocation of `selectVirtualServerByPort(9987)`, which should produce the correct results.

The problem is, though, that a) the code you posted doesn't include any calls to `whoAmI()`, so I can't actually see when you're calling that and b) your log doesn't include the output produced by the `use` command that corresponds to that `selectVirtualServerByPort` call, so I can't judge whether that worked correctly, either.

Short version: Give me more info and I'll see if we can figure out why you're experiencing problems.

**Comment by Delandilucas** [comment]
My code, copied the examples of class ChatBotExample, only difference is the selectVirtualServerByPort () I used!

**Comment by Firedroide** [comment]
Could you please still post your code and the complete log output?
The ones in this issue are not complete enough and don't match up.

I love puzzling, but I don't like doing it with half the pieces missing and another quarter being from an entirely different puzzle.

**Comment by Delandilucas** [comment]
[2016-07-15 22:23:49.068] < TS3
[2016-07-15 22:23:49.070] < 
[2016-07-15 22:23:49.306] < Welcome to the TeamSpeak 3 ServerQuery interface, type "help" for a list of commands and "help <command>" for information on a specific command.
[2016-07-15 22:23:49.306] < 
[2016-07-15 22:23:49.737] > login serveradmin steamwgbrasil
[2016-07-15 22:23:50.114] [login] < error id=0 msg=ok
[2016-07-15 22:23:50.440] > use port=9987
[2016-07-15 22:23:51.397] [use] < error id=0 msg=ok
[2016-07-15 22:23:51.493] > clientupdate client_nickname=MushTest
[2016-07-15 22:23:51.910] [clientupdate] < error id=513 msg=nickname\sis\salready\sin\suse
[2016-07-15 22:23:51.911] [SEVERE] TS3 command error: {msg=nickname is already in use, id=513}
[2016-07-15 22:23:52.195] > sendtextmessage targetmode=2 target=0 msg=MushTest\sestá\sonline!!
[2016-07-15 22:23:52.612] [sendtextmessage] < error id=0 msg=ok
[2016-07-15 22:23:52.895] > whoami
[2016-07-15 22:23:53.312] [whoami] < virtualserver_status=online virtualserver_id=9 virtualserver_unique_identifier=E4fQjkAXwp\/pGLVr5XZd6yNJr9Q= virtualserver_port=9987 client_id=141 client_channel_id=11652 client_nickname=serveradmin\sfrom\s138.99.33.25:50736 client_database_id=1 client_login_name=serveradmin client_unique_identifier=serveradmin client_origin_server_id=0
[2016-07-15 22:23:53.550] [whoami] < error id=0 msg=ok
[2016-07-15 22:23:53.595] > servernotifyregister event=textchannel id=0
[2016-07-15 22:23:54.011] [servernotifyregister] < error id=0 msg=ok


Code:

public static void main(String[] args) {

		final TS3Config config = new TS3Config();
		config.setHost("ts.wgbrasil.tk");
		config.setDebugLevel(Level.ALL);
		
		final TS3Query query = new TS3Query(config);
		query.connect();
		
		final TS3Api api = new TS3Api(query);
		api.login("serveradmin", "steamwgbrasil");
		api.selectVirtualServerByPort(9987);
		api.setNickname("MushTest");
		api.sendChannelMessage("MushTest está online!!");
		
		final int clientId = api.whoAmI().getId();
		
		api.registerEvent(TS3EventType.TEXT_CHANNEL, 0);
		
		api.addTS3Listeners(new TS3EventAdapter() {
			
			@Override
			public void onTextMessage(TextMessageEvent e){ 
				
				if(e.getTargetMode() == TextMessageTargetMode.CHANNEL && e.getInvokerId() != clientId){
					String message = e.getMessage().toLowerCase();
					
					if(message.equals("!ping")){
						api.sendChannelMessage("pong");
					} else if (message.startsWith("hello")){
						api.sendChannelMessage("Hello " + e.getInvokerName() + "!");
					}
				}
			}
		});		
	}

**Comment by mohe2015** [comment]
It says "nickname is already in use" so you are propably running your program several times simultaneously.

**Comment by Firedroide** [comment]
And I'd also like to mention that in the log and code you've sent us now, you're correctly selecting a virtual server. Now you're just trying to use a nickname that another client or query on the server already has. Those have to be unique, though.

But see, that's what I mean by puzzling with missing pieces: You've described this issue as if you were having problems with virtual servers and with the `selectVirtualServerByX` methods, but as it turns out, those actually don't even cause the problem.

**Comment by Delandilucas** [comment]
Thanks for the help, solved!
